File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
n = int(input())
time_list = []

for i in range(n) :
    start, end = map(int, input().split())
    time_list.append([start, end])

time_list.sort(key=lambda x : (x[1], x[0])) #끝나는 시간으로 오름차순 정렬

end_time = 0                                #다음 시작 시간을 찾기 위해 끝나는 시간 저장
count = 0

for start_plan, end_plan in time_list :
    if start_plan >= end_time :             #시작 시간이 끝나는 시간보다 크거나 같은경우
        end_time = end_plan                 #시작 시간의 끝이 다음 끝나는 시간이 된다 
        count += 1

print(count)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.js
let isBaekjoon = 1; //0로컬, 1백준
let fs = require("fs");
let input = isBaekjoon
    ? fs.readFileSync("/dev/stdin").toString().split("\n")
    : fs.readFileSync("./test.txt").toString().split("\n");

let n = Number(input[0]);
let roomList = [];
for (let i = 1; i <= n; i++) {
    roomList.push(input[i].split(" ").map(Number));
}

const sortRoomList = roomList.sort((a, b) => {
    if (a[1] != b[1]) {
        return a[1] - b[1];
    }

    return a[0] - b[0];
});

let count = 0;
let catchRoomTime = 0;

for (let i = 0; i < n; i++) {
    if (sortRoomList[i][0] >= catchRoomTime) {
        catchRoomTime = sortRoomList[i][1];
        count++;
    }
}

console.log(count);

//두번째껄로 오름차순 정렬
//바로 끝나는거 바로 뒤에있는거 수행
